web/search.dart

```diff
@@ -251,21 +298,70 @@ void initializeSearch(
   }
 
   void showSuggestions() {
-    if (presentation.hasChildNodes()) {
+    if (searchResults.hasChildNodes()) {
       listBox.style.display = 'block';
       listBox.setAttribute('aria-expanded', 'true');
     }
   }
 
+  // Function that iterates through the map and appends it to the given HTML element.
+  void iterateCategoriesMap(Element e) {
+    for (var k in categoriesMap.keys) {
+      if (categoriesMap[k] != null) {
+        var values = categoriesMap[k];
+        e.append(values!);
+      }
+    }
+  }
+
+  // Function that creates the content displayed in the main-content element.
+  void searchResultPage(String input) {
+    var mainContent = document.getElementById('dartdoc-main-content');
+    mainContent?.text = '';
```

Comment:
Should we just `return;` if mainContent is null? Save all the work below and simplify the null checks.